test/logic_value_test.dart

```diff
@@ -1252,4 +1252,190 @@ void main() {
       }
     });
   });
+  group('Comparable LogicValue', () {
+    test('positive - int', () {
+      final a = LogicValue.ofInt(3, 8);
+      final b = LogicValue.ofInt(0, 8);
+      final c = LogicValue.ofInt(1, 8);
+      final d = LogicValue.ofInt(2, 8);
+      final e = LogicValue.ofInt(23, 8);
+      final f = LogicValue.ofInt(44, 8);
+      final g = LogicValue.ofInt(2, 8);
+      final h = LogicValue.ofInt(9, 8);
+
+      final values = <LogicValue>[a, b, c, d, e, f, g, h];
+
+      final expected = List<int>.filled(values.length, 0);
+
+      for (var i = 0; i < values.length; i++) {
+        expected[i] = values[i].toInt();
+      }
+
+      values.sort();
+      expected.sort();
+      for (var i = 0; i < values.length; i++) {
+        expect(values[i].toInt(), expected[i]);
+      }
+    });
+    test('unsigned values - int 64 bits', () {
+      final a = LogicValue.ofInt(3, 64);
+      final b = LogicValue.ofInt(0, 64);
+      final c = LogicValue.ofInt(1, 64);
+      final d = LogicValue.ofInt(2, 64);
+      final e = LogicValue.ofInt(23, 64);
+      final f = LogicValue.ofInt(-127, 64);
+      final g = LogicValue.ofInt(-128, 64);
+      final h = LogicValue.ofInt(-1, 64);
+
+      final values = <LogicValue>[a, b, c, d, e, f, g, h];
+
+      final expected = List<BigInt>.filled(values.length, BigInt.zero);
+
+      for (var i = 0; i < values.length; i++) {
+        expected[i] =
+            BigInt.from(values[i].toInt()).toUnsigned(values[i].width);
+      }
+
+      values.sort();
+      expected.sort();
+      for (var i = 0; i < values.length; i++) {
+        expect(BigInt.from(values[i].toInt()).toUnsigned(values[i].width),
+            expected[i]);
+      }
+    });
+    test('unsigned values - int 8 bits', () {
+      final a = LogicValue.ofInt(3, 8);
+      final b = LogicValue.ofInt(0, 8);
+      final c = LogicValue.ofInt(1, 8);
+      final d = LogicValue.ofInt(2, 8);
+      final e = LogicValue.ofInt(23, 8);
+      final f = LogicValue.ofInt(-127, 8);
+      final g = LogicValue.ofInt(-128, 8);
+      final h = LogicValue.ofInt(-1, 8);
+
+      final values = <LogicValue>[a, b, c, d, e, f, g, h];
+
+      final expected = List<BigInt>.filled(values.length, BigInt.zero);
+
+      for (var i = 0; i < values.length; i++) {
+        expected[i] =
+            BigInt.from(values[i].toInt()).toUnsigned(values[i].width);
+      }
+
+      values.sort();
+      expected.sort();
+
+      for (var i = 0; i < values.length; i++) {
+        expect(BigInt.from(values[i].toInt()).toUnsigned(values[i].width),
+            expected[i]);
+      }
+    });
+    test('unsigned  BigInt(invalid int)  & int ', () {
+      final a = LogicValue.ofBigInt(BigInt.parse('3'), 64);
+      final b = LogicValue.ofBigInt(BigInt.zero, 64);
+      final c = LogicValue.ofBigInt(BigInt.from(-1), 64); //Invalid Int
```

Comment:
what makes this "invalid"?

Reply:
Since `-1 BigInt` of 64 bits is an `invalid` int and we are comparing `BigInt` and `int`, so
 I kept sanity checks as I was accepting the signed number input in the `-2^(n-1) to 2^(n-1)-1` range, but processing it like an unsigned int.

Since we are dealing only with unsigned values, maybe it is redundant.